### PR TITLE
Devex - use docketNumber instead of caseId

### PR DESCRIPTION
**Pull request by rachelschneiderman**
- UserCaseNote entity
- getUserCaseNoteLambda
- updateUserCaseNoteLambda
- deleteUserCaseNoteLambda

### CODE REVIEW

File: shared/src/business/entities/notes/UserCaseNote.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ function UserCaseNote(rawProps) {
   this.entityName = 'UserCaseNote';
 
   this.caseId = rawProps.caseId;
```
Comment (rachaelparris) [inline shared/src/business/entities/notes/UserCaseNote.js]: Can `caseId` be removed from this entity now?

Comment (rachelschneiderman) [reply shared/src/business/entities/notes/UserCaseNote.js]: I believe so!

File: shared/src/business/useCases/caseNote/deleteUserCaseNoteInteractor.test.js [hunk 2]
```diff
@@ -32,30 +27,20 @@ describe('deleteUserCaseNoteInteractor', () => {
...
       getJudgeForUserChambersInteractor: () => ({
         role: ROLES.judge,
         userId: '6805d1ab-18d0-43ec-bafb-654e83405416',
       }),
     });
 
-    let error;
-    let caseNote;
-
-    try {
```
Comment (rachaelparris) [inline shared/src/business/useCases/caseNote/deleteUserCaseNoteInteractor.test.js:49]: 🎉

File: shared/src/business/useCases/caseNote/updateUserCaseNoteInteractor.js [hunk 3]
```diff
@@ -28,8 +29,17 @@ exports.updateUserCaseNoteInteractor = async ({
...
+    .getPersistenceGateway()
+    .getCaseByDocketNumber({
+      applicationContext,
+      docketNumber,
+    });
+  const caseEntity = new Case(caseRecord, { applicationContext });
+
   const caseNoteEntity = new UserCaseNote({
-    caseId,
+    caseId: caseEntity.caseId,
```
Comment (rachaelparris) [inline shared/src/business/useCases/caseNote/updateUserCaseNoteInteractor.js]: I think when you remove `caseId` from the entity, this should no longer be needed, along with the persistence call above.

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**